### ISSUE TITLE
ssm: discretise_mfd — matrix-fraction discretisation for learned or degenerate F

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#140]
## Problem / Request

`SDEKernel.discretise` computes `Q = P_inf − A P_inf Aᵀ`, which requires a
subclass-supplied stationary covariance `P_inf`. There is no discretisation path
for an `F` that is a **learned parameter**, where `P_inf` does not exist as an
attribute and must be recovered from the Lyapunov equation — an equation that is
singular for an important class of `F`.

Add `discretise_mfd(F, Q_c, dt)`, which computes both `A` and `Q` from a single
`2n × 2n` matrix exponential and is well-defined for **every** `F`.

## User Story

> As a user fitting a continuous-time state-space model whose drift matrix is
> learned rather than derived from a kernel, I want a discretisation routine that
> does not require a stationary covariance, so that my model does not silently
> produce a wrong process-noise matrix when the drift acquires an undamped
> oscillatory mode.

## Motivation

For the SDE kernels currently in the repo this adds nothing — the two routes agree
to 2.9e-15 on Matérn-3/2 across `dt ∈ {0.01, 0.5, 3.0}`. The gap opens as soon as
`F` is fitted.

The Lyapunov equation `F P + P Fᵀ + Q_c = 0` is uniquely solvable **iff no two
eigenvalues of `F` sum to zero**. Measured (`cond_test.py`):

| eigenvalues of `F` | Sylvester residual | `‖Q_Lyap − Q_true‖∞` | `‖Q_MFD − Q_true‖∞` |
|---|---|---|---|
| (0.3, 0.1) | 8.9e-16 | 1.5e-11 | 1.5e-11 |
| (0.3, −0.29) | 0.0 | 1.1e-11 | 1.1e-11 |
| (0.3, −0.2999) | 9.1e-13 | 1.5e-11 | 1.1e-11 |
| **(0.3, −0.3)** | **3.1e-01** | **2.46** | **1.1e-11** |
| **(±1.4i)** | **1.0** | **4.9e-01** | **1.5e-11** |

The last row is not a corner case. An **undamped oscillatory mode has
`λ = ±iω`, so `λ + λ̄ = 0` always.** `CosineSDE` has exactly that drift
(`src/gaussx/_ssm/_periodic.py:37`: `F = [[0, -w], [w, 0]]`, `Q_c = 0`), and both
`CosineSDE` and `PeriodicSDE` sidestep the problem by overriding `discretise`
with closed forms. That fix is correct and unavailable when `F` is learned.

## Proposed API

```python
def discretise_mfd(
    F: Float[Array, "d d"],
    Q_c: Float[Array, "d d"],
    dt: Float[Array, ""],
) -> tuple[Float[Array, "d d"], Float[Array, "d d"]]:
    r"""Discretise a linear SDE by matrix-fraction decomposition.

    Returns $(A, Q)$ for the discrete-time model $x_k = A x_{k-1} + q_k$ with
    $q_k \sim N(0, Q)$, where

    $$
    A = e^{F\,\Delta t}, \qquad
    Q = \int_0^{\Delta t} e^{Fs}\, Q_c\, e^{F^\top s}\, ds .
    $$

    Both are recovered from a single $2d \times 2d$ matrix exponential.
    Unlike the stationary route in `SDEKernel.discretise`, this needs no
    $P_\infty$ and is well-defined for every $F$ -- including a learned $F$
    whose eigenvalues sum to zero in pairs, where the Lyapunov equation
    $F P + P F^\top + Q_c = 0$ has no unique solution. Undamped oscillatory
    modes ($\lambda = \pm i\omega$) always fall in that case.

    Args:
        F: Continuous-time drift matrix, shape ``(d, d)``.
        Q_c: Continuous-time diffusion covariance $L Q_c L^\top$, shape
            ``(d, d)``.
        dt: Time step. Must be non-negative.

    Returns:
        Tuple ``(A, Q)``, both shape ``(d, d)``. ``Q`` is symmetrised.
    """


def discretise_mfd_sequence(
    F: Float[Array, "d d"],
    Q_c: Float[Array, "d d"],
    dt: Float[Array, " N"],
) -> tuple[Float[Array, "N d d"], Float[Array, "N d d"]]:
    """Vectorised `discretise_mfd` over a vector of time steps."""
```

## Design Snapshot

```python
def discretise_mfd(F, Q_c, dt):
    d = F.shape[0]
    Phi = jnp.block([[F, Q_c], [jnp.zeros((d, d), F.dtype), -F.T]])
    M = jsl.expm(Phi * dt)
    C, D = M[:d, d:], M[d:, d:]
    A = jsl.expm(F * dt)
    Q = C @ jnp.linalg.inv(D)          # D = expm(-Fᵀ dt), always invertible
    return A, symmetrize(Q)
```

Wiring into the kernel surface — a fallback, not a replacement, so the analytic
closed forms in `MaternSDE` / `CosineSDE` / `PeriodicSDE` keep their precision:

```python
# SDEKernel.discretise, src/gaussx/_ssm/_sde_kernel.py:60
params = self.sde_params()
if params.P_inf is None:
    return discretise_mfd(params.F, params.L @ params.Q_c @ params.L.T, dt)
A = jsl.expm(params.F * dt)
Q = symmetrize(params.P_inf - A @ params.P_inf @ A.T)
return A, Q
```

This requires `SDEParams.P_inf` to become `Float[Array, "d d"] | None`. Every
existing subclass supplies it, so the fallback is dead code today and live the
moment someone writes a `LearnedSDE`.

## Mathematical Notes

```text
Van Loan (1978). For the augmented generator

        Φ = [  F   Q_c ]
            [  0  -Fᵀ  ]

        exp(Φ·Δt) = [ M₁₁  C ]      with   A = M₁₁ = e^{FΔt}
                    [  0   D ]             Q = C·D⁻¹
                                           D = e^{-FᵀΔt}   (always invertible)

Relation to the stationary route. If P solves the Lyapunov equation
F P + P Fᵀ + Q_c = 0, then

        Q(Δt) = P − A P Aᵀ                                        (†)

CORRECTION worth stating in the docstring: (†) does NOT require F to be stable,
and it does NOT require P to be PSD. Measured on an unstable F with eigenvalues
(0.3, 0.1), the Lyapunov solution has eigenvalues [−9.87, −0.295] — not a valid
covariance — yet (†) still reproduces the true integral to 5.9e-11.

The obstruction is DEGENERACY of the Sylvester system, not instability. Do not
"fix" this by constraining F to be Hurwitz; that would forbid exactly the
oscillatory modes the feature exists to support.

Uniqueness condition:  λᵢ(F) + λⱼ(F) ≠ 0  for all i, j.
```

## References & Existing Code

- Design doc: `kalman_survae_report.md` §2.3, §4-B
- Reference impl (PyTorch): [`johannaSommer/KF_irreg_TS`](https://github.com/johannaSommer/KF_irreg_TS) `models/KF_base.py:48-79` — correct algebra, but builds `M` on CPU and `.to(device)`s afterwards; do not carry that over
- Van Loan, C. F. (1978), *Computing integrals involving the matrix exponential*, IEEE TAC 23(3)
- Existing stationary route: `src/gaussx/_ssm/_sde_kernel.py:60-98`
- The subclasses that already work around this: `src/gaussx/_ssm/_periodic.py:41-53` (`CosineSDE.discretise`), `:106` (`PeriodicSDE.discretise`)

## Implementation Steps

- [ ] Add `discretise_mfd` and `discretise_mfd_sequence` in a new `src/gaussx/_ssm/_discretise.py`
- [ ] Relax `SDEParams.P_inf` to `Float[Array, "d d"] | None` in `src/gaussx/_ssm/_sde_kernel.py:16`
- [ ] Add the `P_inf is None` fallback branch to `SDEKernel.discretise`
- [ ] Export both from `src/gaussx/_ssm/__init__.py` and `src/gaussx/__init__.py`
- [ ] Add both to the `members:` list in `docs/api/ssm.md` under "SDE kernels" — `tests/test_docs_api_coverage.py` fails otherwise
- [ ] Guard `dt < 0` with a clear error rather than a silent `expm` of a negative step

## Definition of Done

- [ ] Code lands at the intended path
- [ ] Public API exported via `src/gaussx/__init__.py`
- [ ] Tests pass: `make test`
- [ ] Lint + typecheck pass: `make lint && make typecheck`
- [ ] Docstrings (Google-style, `$...$` math, no Sphinx/RST) on all public symbols

## Testing

New file `tests/ssm/test_discretise_mfd.py`.

- [ ] Agreement test: MFD vs the `P_inf` route on `MaternSDE(nu=1.5)` for `dt ∈ {0.01, 0.5, 3.0}` — **target ≤ 1e-13** (measured 6.66e-16 / 4.44e-16 / 2.89e-15)
- [ ] Ground-truth test: MFD vs 40k-point trapezoid quadrature of `∫₀^Δt e^{Fs} Q_c e^{Fᵀs} ds` on an **unstable** `F` (eigenvalues 0.3, 0.1) — **target ≤ 1e-9** (measured 5.9e-11)
- [ ] **The load-bearing test:** on the undamped oscillator `F = [[0, −ω], [ω, 0]]`, `ω = 1.4`, MFD matches quadrature to ≤ 1e-9 (measured 1.48e-11) while `scipy.linalg.solve_sylvester` returns a solution with residual ≈ 1.0 and error ≈ 4.9e-01. Assert the MFD side; assert the Lyapunov side *fails* so the test documents why the function exists
- [ ] Degeneracy sweep: parameterise over `λ₂ ∈ {−0.29, −0.2999, −0.3}` with `λ₁ = 0.3` and assert MFD error stays ≤ 1e-9 across all three while the Lyapunov route blows up only at `−0.3`
- [ ] PSD test: `Q` returned by MFD has non-negative eigenvalues for every case above
- [ ] Symmetry test: `Q == Q.T` exactly (the `symmetrize` call is not optional — `C @ inv(D)` is not symmetric to floating point)
- [ ] Consistency test: `discretise_mfd(F, Q_c, 0.0)` returns `(I, 0)`
- [ ] Property test: `discretise_mfd_sequence` matches a `vmap` of the scalar version
- [ ] Regression test: `MaternSDE`, `CosineSDE`, `PeriodicSDE`, `ConstantSDE` still take their analytic paths — assert `discretise` output is unchanged bit-for-bit against a stored reference, so the fallback cannot silently capture them
- [ ] `jax.grad` through `discretise_mfd` w.r.t. `F` returns finite values

## Documentation

- [ ] `docs/api/ssm.md` — new prose paragraph in the SDE-kernel section explaining when to reach for MFD over the stationary route, with the `λᵢ + λⱼ = 0` condition stated explicitly
- [ ] Docstrings (covered by Definition of Done)

## Relationships

- Parent (theme epic): #
- Blocked by: #
- Blocks: #
- Related: # (gaussx: per-dimension mask), # (gauss_flows: LGSSM base distribution)
